src/tvgFlutterLottieAnimation.cpp

```diff
@@ -0,0 +1,311 @@
+#include <thorvg.h>
+#include "tvgFlutterLottieAnimation.h"
+
+using namespace std;
+using namespace tvg;
+
+// NOTE: build error with `const char*`
+static char *NoError = "None";
+
+class __attribute__((visibility("default"))) TvgLottieAnimation
+{
+public:
+    ~TvgLottieAnimation()
+    {
+        free(buffer);
+        Initializer::term(CanvasEngine::Sw);
+    }
+
+    static unique_ptr<TvgLottieAnimation> create()
+    {
+        return unique_ptr<TvgLottieAnimation>(new TvgLottieAnimation());
+    }
+
+    char *error()
+    {
+        return errorMsg;
+    }
+
+    float *size()
+    {
+        return psize;
+    }
+
+    float duration()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->duration();
+    }
+
+    float totalFrame()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->totalFrame();
+    }
+
+    float curFrame()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->curFrame();
+    }
+
+    void resize(int w, int h)
+    {
+        if (!canvas || !animation)
+            return;
+        if (width == w && height == h)
+            return;
+
+        width = w;
+        height = h;
+
+        free(buffer);
+        buffer = (uint8_t *)malloc(width * height * sizeof(uint32_t));
+        canvas->target((uint32_t *)buffer, width, width, height, SwCanvas::ABGR8888S);
+
+        float scale;
+        float shiftX = 0.0f, shiftY = 0.0f;
+        if (psize[0] > psize[1])
+        {
+            scale = width / psize[0];
+            shiftY = (height - psize[1] * scale) * 0.5f;
+        }
+        else
+        {
+            scale = height / psize[1];
+            shiftX = (width - psize[0] * scale) * 0.5f;
+        }
+        animation->picture()->scale(scale);
+        animation->picture()->translate(shiftX, shiftY);
+
+        updated = true;
+    }
+
+    bool load(char *data, char *mimetype, int width, int height)
+    {
+        errorMsg = NoError;
+
+        if (!canvas)
+            return 0;
+
+        if (data != NULL && data[0] == '\0')
+        {
+            errorMsg = "Invalid data";
+            return 0;
+        }
+
+        canvas->clear(true);
+
+        animation = Animation::gen();
+
+        auto result = animation->picture()->load(data, sizeof(data), "lottie", false);
+
+        if (result != Result::Success)
+        {
+            switch (result)
```

Comment:
some separate fun? 
also 'd standardize the error messages -currently, some are like those from tvg, while others are custom